File: PST/FUEL Source Code/FUEL/dlgTestStationType.cs
// Decompiled with JetBrains decompiler
// Type: FUEL.dlgTestStationType
// Assembly: FUEL, Version=2.4.8.0, Culture=neutral, PublicKeyToken=null
// MVID: A3D630D0-0620-425D-8CFB-B67B2AC75507
// Assembly location: C:\hp\FUEL\FUEL.dll

using DevComponents.DotNetBar;
using DevComponents.DotNetBar.Controls;
using DevComponents.DotNetBar.Metro;
using FUEL.My;
using Microsoft.VisualBasic;
using Microsoft.VisualBasic.CompilerServices;
using System;
using System.Collections;
using System.Collections.Generic;
using System.ComponentModel;
using System.Diagnostics;
using System.Drawing;
using System.Runtime.CompilerServices;
using System.Windows.Forms;

namespace FUEL
{
  [DesignerGenerated]
  public class dlgTestStationType : MetroForm
  {
    private static List<WeakReference> __ENCList = new List<WeakReference>();
    private IContainer components;
    [AccessedThroughProperty("TableLayoutPanel1")]
    private TableLayoutPanel _TableLayoutPanel1;
    [AccessedThroughProperty("OK_Button")]
    private Button _OK_Button;
    [AccessedThroughProperty("Cancel_Button")]
    private Button _Cancel_Button;
    [AccessedThroughProperty("cboTestStations")]
    private ComboBoxEx _cboTestStations;
    [AccessedThroughProperty("LabelX1")]
    private LabelX _LabelX1;

    [DebuggerNonUserCode]
    static dlgTestStationType()
    {
    }

    [DebuggerNonUserCode]
    public dlgTestStationType()
    {
      this.Load += new EventHandler(this.dlgTestStationType_Load);
      dlgTestStationType.__ENCAddToList((object) this);
      this.InitializeComponent();
    }

    [DebuggerNonUserCode]
    private static void __ENCAddToList(object value)
    {
      lock (dlgTestStationType.__ENCList)
      {
        if (dlgTestStationType.__ENCList.Count == dlgTestStationType.__ENCList.Capacity)
        {
          int index1 = 0;
          int num = checked (dlgTestStationType.__ENCList.Count - 1);
          int index2 = 0;
          while (index2 <= num)
          {
            if (dlgTestStationType.__ENCList[index2].IsAlive)
            {
              if (index2 != index1)
                dlgTestStationType.__ENCList[index1] = dlgTestStationType.__ENCList[index2];
              checked { ++index1; }
            }
            checked { ++index2; }
          }
          dlgTestStationType.__ENCList.RemoveRange(index1, checked (dlgTestStationType.__ENCList.Count - index1));
          dlgTestStationType.__ENCList.Capacity = dlgTestStationType.__ENCList.Count;
        }
        dlgTestStationType.__ENCList.Add(new WeakReference(RuntimeHelpers.GetObjectValue(value)));
      }
    }

    [DebuggerNonUserCode]
    protected override void Dispose(bool disposing)
    {
      try
      {
        if (!disposing || this.components == null)
          return;
        this.components.Dispose();
      }
      finally
      {
        base.Dispose(disposing);
      }
    }

    [DebuggerStepThrough]
    private void InitializeComponent()
    {
      this.TableLayoutPanel1 = new TableLayoutPanel();
      this.OK_Button = new Button();
      this.Cancel_Button = new Button();
      this.cboTestStations = new ComboBoxEx();
      this.LabelX1 = new LabelX();
      this.TableLayoutPanel1.SuspendLayout();
      this.SuspendLayout();
      this.TableLayoutPanel1.Anchor = AnchorStyles.Bottom | AnchorStyles.Right;
      this.TableLayoutPanel1.BackColor = Color.Transparent;
      this.TableLayoutPanel1.ColumnCount = 2;
      this.TableLayoutPanel1.ColumnStyles.Add(new ColumnStyle(SizeType.Percent, 50f));
      this.TableLayoutPanel1.ColumnStyles.Add(new ColumnStyle(SizeType.Percent, 50f));
      this.TableLayoutPanel1.Controls.Add((Control) this.OK_Button, 0, 0);
      this.TableLayoutPanel1.Controls.Add((Control) this.Cancel_Button, 1, 0);
      this.TableLayoutPanel1.ForeColor = Color.Black;
      TableLayoutPanel tableLayoutPanel1_1 = this.TableLayoutPanel1;
      Point point1 = new Point(165, 74);
      Point point2 = point1;
      tableLayoutPanel1_1.Location = point2;
      this.TableLayoutPanel1.Name = "TableLayoutPanel1";
      this.TableLayoutPanel1.RowCount = 1;
      this.TableLayoutPanel1.RowStyles.Add(new RowStyle(SizeType.Percent, 50f));
      TableLayoutPanel tableLayoutPanel1_2 = this.TableLayoutPanel1;
      Size size1 = new Size(146, 29);
      Size size2 = size1;
      tableLayoutPanel1_2.Size = size2;
      this.TableLayoutPanel1.TabIndex = 0;
      this.OK_Button.Anchor = AnchorStyles.None;
      this.OK_Button.BackColor = Color.White;
      this.OK_Button.ForeColor = Color.Black;
      Button okButton1 = this.OK_Button;
      point1 = new Point(3, 3);
      Point point3 = point1;
      okButton1.Location = point3;
      this.OK_Button.Name = "OK_Button";
      Button okButton2 = this.OK_Button;
      size1 = new Size(67, 23);
      Size size3 = size1;
      okButton2.Size = size3;
      this.OK_Button.TabIndex = 0;
      this.OK_Button.Text = "OK";
      this.OK_Button.UseVisualStyleBackColor = false;
      this.Cancel_Button.Anchor = AnchorStyles.None;
      this.Cancel_Button.BackColor = Color.White;
      this.Cancel_Button.DialogResult = DialogResult.Cancel;
      this.Cancel_Button.ForeColor = Color.Black;
      Button cancelButton1 = this.Cancel_Button;
      point1 = new Point(76, 3);
      Point point4 = point1;
      cancelButton1.Location = point4;
      this.Cancel_Button.Name = "Cancel_Button";
      Button cancelButton2 = this.Cancel_Button;
      size1 = new Size(67, 23);
      Size size4 = size1;
      cancelButton2.Size = size4;
      this.Cancel_Button.TabIndex = 1;
      this.Cancel_Button.Text = "Cancel";
      this.Cancel_Button.UseVisualStyleBackColor = false;
      this.cboTestStations.DisplayMember = "Text";
      this.cboTestStations.DrawMode = DrawMode.OwnerDrawFixed;
      this.cboTestStations.DropDownStyle = ComboBoxStyle.DropDownList;
      this.cboTestStations.ForeColor = Color.Black;
      this.cboTestStations.FormattingEnabled = true;
      this.cboTestStations.ItemHeight = 14;
      ComboBoxEx cboTestStations1 = this.cboTestStations;
      point1 = new Point(13, 42);
      Point point5 = point1;
      cboTestStations1.Location = point5;
      this.cboTestStations.Name = "cboTestStations";
      ComboBoxEx cboTestStations2 = this.cboTestStations;
      size1 = new Size(203, 20);
      Size size5 = size1;
      cboTestStations2.Size = size5;
      this.cboTestStations.Style = eDotNetBarStyle.StyleManagerControlled;
      this.cboTestStations.TabIndex = 1;
      this.LabelX1.Anchor = AnchorStyles.Top | AnchorStyles.Left | AnchorStyles.Right;
      this.LabelX1.BackColor = Color.White;
      this.LabelX1.BackgroundStyle.CornerType = eCornerType.Square;
      this.LabelX1.ForeColor = Color.Black;
      LabelX labelX1_1 = this.LabelX1;
      point1 = new Point(13, 13);
      Point point6 = point1;
      labelX1_1.Location = point6;
      this.LabelX1.Name = "LabelX1";
      LabelX labelX1_2 = this.LabelX1;
      size1 = new Size(298, 23);
      Size size6 = size1;
      labelX1_2.Size = size6;
      this.LabelX1.TabIndex = 2;
      this.LabelX1.Text = "Please select the type of test station that you are working at.";
      this.AcceptButton = (IButtonControl) this.OK_Button;
      this.AutoScaleDimensions = new SizeF(6f, 13f);
      this.AutoScaleMode = AutoScaleMode.Font;
      this.CancelButton = (IButtonControl) this.Cancel_Button;
      size1 = new Size(323, 115);
      this.ClientSize = size1;
      this.Controls.Add((Control) this.LabelX1);
      this.Controls.Add((Control) this.cboTestStations);
      this.Controls.Add((Control) this.TableLayoutPanel1);
      this.DoubleBuffered = true;
      this.FormBorderStyle = FormBorderStyle.FixedDialog;
      this.MaximizeBox = false;
      this.MinimizeBox = false;
      this.Name = nameof (dlgTestStationType);
      this.ShowInTaskbar = false;
      this.StartPosition = FormStartPosition.CenterParent;
      this.Text = "Test Station Selection";
      this.TableLayoutPanel1.ResumeLayout(false);
      this.ResumeLayout(false);
    }

    internal virtual TableLayoutPanel TableLayoutPanel1
    {
      [DebuggerNonUserCode] get => this._TableLayoutPanel1;
      [DebuggerNonUserCode, MethodImpl(MethodImplOptions.Synchronized)] set => this._TableLayoutPanel1 = value;
    }

    internal virtual Button OK_Button
    {
      [DebuggerNonUserCode] get => this._OK_Button;
      [DebuggerNonUserCode, MethodImpl(MethodImplOptions.Synchronized)] set
      {
        EventHandler eventHandler = new EventHandler(this.OK_Button_Click);
        if (this._OK_Button != null)
          this._OK_Button.Click -= eventHandler;
        this._OK_Button = value;
        if (this._OK_Button == null)
          return;
        this._OK_Button.Click += eventHandler;
      }
    }

    internal virtual Button Cancel_Button
    {
      [DebuggerNonUserCode] get => this._Cancel_Button;
      [DebuggerNonUserCode, MethodImpl(MethodImplOptions.Synchronized)] set
      {
        EventHandler eventHandler = new EventHandler(this.Cancel_Button_Click);
        if (this._Cancel_Button != null)
          this._Cancel_Button.Click -= eventHandler;
        this._Cancel_Button = value;
        if (this._Cancel_Button == null)
          return;
        this._Cancel_Button.Click += eventHandler;
      }
    }

    internal virtual ComboBoxEx cboTestStations
    {
      [DebuggerNonUserCode] get => this._cboTestStations;
      [DebuggerNonUserCode, MethodImpl(MethodImplOptions.Synchronized)] set
      {
        EventHandler eventHandler = new EventHandler(this.cboTestStations_SelectedIndexChanged);
        if (this._cboTestStations != null)
          this._cboTestStations.SelectedIndexChanged -= eventHandler;
        this._cboTestStations = value;
        if (this._cboTestStations == null)
          return;
        this._cboTestStations.SelectedIndexChanged += eventHandler;
      }
    }

    internal virtual LabelX LabelX1
    {
      [DebuggerNonUserCode] get => this._LabelX1;
      [DebuggerNonUserCode, MethodImpl(MethodImplOptions.Synchronized)] set => this._LabelX1 = value;
    }

    private PST.TestStationTypes _TestStationType { [DebuggerNonUserCode] get; [DebuggerNonUserCode] set; }

    public PST.TestStationTypes TestStationType => this._TestStationType;

    private void OK_Button_Click(object sender, EventArgs e)
    {
      this.DialogResult = DialogResult.OK;
      this._TestStationType = (PST.TestStationTypes) this.cboTestStations.SelectedIndex;
      this.Close();
    }

    private void Cancel_Button_Click(object sender, EventArgs e)
    {
      this.DialogResult = DialogResult.Cancel;
      this._TestStationType = ~PST.TestStationTypes.ProductionLine;
      this.Close();
    }

    private void dlgTestStationType_Load(object sender, EventArgs e)
    {
      try
      {
        foreach (object obj in Enum.GetValues(typeof (PST.TestStationTypes)))
          this.cboTestStations.Items.Add((object) RuntimeHelpers.GetObjectValue(obj).ToString());
      }
      finally
      {
        IEnumerator enumerator;
        if (enumerator is IDisposable)
          (enumerator as IDisposable).Dispose();
      }
      if (DateAndTime.DateDiff(DateInterval.Hour, MySettingsProperty.Settings.TestStationType_Date, DateAndTime.Now) >= 6L)
      {
        MySettingsProperty.Settings.TestStationType = -1;
        MySettingsProperty.Settings.Save();
      }
      else
      {
        this._TestStationType = (PST.TestStationTypes) MySettingsProperty.Settings.TestStationType;
        this.DialogResult = DialogResult.OK;
        this.Close();
      }
      this.cboTestStations.SelectedIndex = MySettingsProperty.Settings.TestStationType;
    }

    private void cboTestStations_SelectedIndexChanged(object sender, EventArgs e)
    {
      MySettingsProperty.Settings.TestStationType = this.cboTestStations.SelectedIndex;
      MySettingsProperty.Settings.TestStationType_Date = DateAndTime.Now;
      MySettingsProperty.Settings.Save();
    }
  }
}

File: PST/FUEL/src/FUEL/dlgTestStationType.vb
Imports DevComponents.DotNetBar
Imports DevComponents.DotNetBar.Controls
Imports DevComponents.DotNetBar.Metro
Imports FUEL.My
Imports Microsoft.VisualBasic
Imports Microsoft.VisualBasic.CompilerServices
Imports System
Imports System.Collections
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Diagnostics
Imports System.Drawing
Imports System.Runtime.CompilerServices
Imports System.Windows.Forms

Namespace FUEL
    <DesignerGenerated> _
    Public Class dlgTestStationType
        Inherits MetroForm
        ' Methods
        <DebuggerNonUserCode> _
        Public Sub New()
            AddHandler MyBase.Load, New EventHandler(AddressOf Me.dlgTestStationType_Load)
            dlgTestStationType.__ENCAddToList(Me)
            Me.InitializeComponent
        End Sub

        <DebuggerNonUserCode> _
        Private Shared Sub __ENCAddToList(ByVal value As Object)
            SyncLock dlgTestStationType.__ENCList
                If (dlgTestStationType.__ENCList.Count = dlgTestStationType.__ENCList.Capacity) Then
                    Dim index As Integer = 0
                    Dim num3 As Integer = (dlgTestStationType.__ENCList.Count - 1)
                    Dim num2 As Integer = 0
                    Do While True
                        Dim num4 As Integer = num3
                        If (num2 > num4) Then
                            dlgTestStationType.__ENCList.RemoveRange(index, (dlgTestStationType.__ENCList.Count - index))
                            dlgTestStationType.__ENCList.Capacity = dlgTestStationType.__ENCList.Count
                            Exit Do
                        End If
                        Dim reference As WeakReference = dlgTestStationType.__ENCList(num2)
                        If reference.IsAlive Then
                            If (num2 <> index) Then
                                dlgTestStationType.__ENCList(index) = dlgTestStationType.__ENCList(num2)
                            End If
                            index += 1
                        End If
                        num2 += 1
                    Loop
                End If
                dlgTestStationType.__ENCList.Add(New WeakReference(value))
            End SyncLock
        End Sub

        Private Sub Cancel_Button_Click(ByVal sender As Object, ByVal e As EventArgs)
            Me.DialogResult = DialogResult.Cancel
            Me._TestStationType = Not TestStationTypes.ProductionLine
            Me.Close
        End Sub

        Private Sub cboTestStations_SelectedIndexChanged(ByVal sender As Object, ByVal e As EventArgs)
            MySettingsProperty.Settings.TestStationType = Me.cboTestStations.SelectedIndex
            MySettingsProperty.Settings.TestStationType_Date = DateAndTime.Now
            MySettingsProperty.Settings.Save
        End Sub

        <DebuggerNonUserCode> _
        Protected Overrides Sub Dispose(ByVal disposing As Boolean)
            Try 
                If (If((Not disposing OrElse (Me.components Is Nothing)), 0, 1) <> 0) Then
                    Me.components.Dispose
                End If
            Finally
                MyBase.Dispose(disposing)
            End Try
        End Sub

        Private Sub dlgTestStationType_Load(ByVal sender As Object, ByVal e As EventArgs)
            Dim enumerator As IEnumerator
            Try 
                enumerator = Enum.GetValues(GetType(TestStationTypes)).GetEnumerator
                Do While True
                    If Not enumerator.MoveNext Then
                        Exit Do
                    End If
                    Dim current As Object = enumerator.Current
                    Me.cboTestStations.Items.Add(current.ToString)
                Loop
            Finally
                If Not Object.ReferenceEquals(TryCast(enumerator,IDisposable), Nothing) Then
                    TryCast(enumerator,IDisposable).Dispose
                End If
            End Try
            If (DateAndTime.DateDiff(DateInterval.Hour, MySettingsProperty.Settings.TestStationType_Date, DateAndTime.Now, FirstDayOfWeek.Sunday, FirstWeekOfYear.Jan1) >= 6) Then
                MySettingsProperty.Settings.TestStationType = -1
                MySettingsProperty.Settings.Save
            Else
                Me._TestStationType = DirectCast(MySettingsProperty.Settings.TestStationType, TestStationTypes)
                Me.DialogResult = DialogResult.OK
                Me.Close
            End If
            Me.cboTestStations.SelectedIndex = MySettingsProperty.Settings.TestStationType
        End Sub

        <DebuggerStepThrough> _
        Private Sub InitializeComponent()
            Me.TableLayoutPanel1 = New TableLayoutPanel
            Me.OK_Button = New Button
            Me.Cancel_Button = New Button
            Me.cboTestStations = New ComboBoxEx
            Me.LabelX1 = New LabelX
            Me.TableLayoutPanel1.SuspendLayout
            Me.SuspendLayout
            Me.TableLayoutPanel1.Anchor = (AnchorStyles.Right Or AnchorStyles.Bottom)
            Me.TableLayoutPanel1.BackColor = Color.Transparent
            Me.TableLayoutPanel1.ColumnCount = 2
            Me.TableLayoutPanel1.ColumnStyles.Add(New ColumnStyle(SizeType.Percent, 50!))
            Me.TableLayoutPanel1.ColumnStyles.Add(New ColumnStyle(SizeType.Percent, 50!))
            Me.TableLayoutPanel1.Controls.Add(Me.OK_Button, 0, 0)
            Me.TableLayoutPanel1.Controls.Add(Me.Cancel_Button, 1, 0)
            Me.TableLayoutPanel1.ForeColor = Color.Black
            Dim point2 As New Point(&HA5, &H4A)
            Me.TableLayoutPanel1.Location = point2
            Me.TableLayoutPanel1.Name = "TableLayoutPanel1"
            Me.TableLayoutPanel1.RowCount = 1
            Me.TableLayoutPanel1.RowStyles.Add(New RowStyle(SizeType.Percent, 50!))
            Dim size2 As New Size(&H92, &H1D)
            Me.TableLayoutPanel1.Size = size2
            Me.TableLayoutPanel1.TabIndex = 0
            Me.OK_Button.Anchor = AnchorStyles.None
            Me.OK_Button.BackColor = Color.White
            Me.OK_Button.ForeColor = Color.Black
            point2 = New Point(3, 3)
            Me.OK_Button.Location = point2
            Me.OK_Button.Name = "OK_Button"
            size2 = New Size(&H43, &H17)
            Me.OK_Button.Size = size2
            Me.OK_Button.TabIndex = 0
            Me.OK_Button.Text = "OK"
            Me.OK_Button.UseVisualStyleBackColor = False
            Me.Cancel_Button.Anchor = AnchorStyles.None
            Me.Cancel_Button.BackColor = Color.White
            Me.Cancel_Button.DialogResult = DialogResult.Cancel
            Me.Cancel_Button.ForeColor = Color.Black
            point2 = New Point(&H4C, 3)
            Me.Cancel_Button.Location = point2
            Me.Cancel_Button.Name = "Cancel_Button"
            size2 = New Size(&H43, &H17)
            Me.Cancel_Button.Size = size2
            Me.Cancel_Button.TabIndex = 1
            Me.Cancel_Button.Text = "Cancel"
            Me.Cancel_Button.UseVisualStyleBackColor = False
            Me.cboTestStations.DisplayMember = "Text"
            Me.cboTestStations.DrawMode = DrawMode.OwnerDrawFixed
            Me.cboTestStations.DropDownStyle = ComboBoxStyle.DropDownList
            Me.cboTestStations.ForeColor = Color.Black
            Me.cboTestStations.FormattingEnabled = True
            Me.cboTestStations.ItemHeight = 14
            point2 = New Point(13, &H2A)
            Me.cboTestStations.Location = point2
            Me.cboTestStations.Name = "cboTestStations"
            size2 = New Size(&HCB, 20)
            Me.cboTestStations.Size = size2
            Me.cboTestStations.Style = eDotNetBarStyle.StyleManagerControlled
            Me.cboTestStations.TabIndex = 1
            Me.LabelX1.Anchor = (AnchorStyles.Right Or (AnchorStyles.Left Or AnchorStyles.Top))
            Me.LabelX1.BackColor = Color.White
            Me.LabelX1.BackgroundStyle.CornerType = eCornerType.Square
            Me.LabelX1.ForeColor = Color.Black
            point2 = New Point(13, 13)
            Me.LabelX1.Location = point2
            Me.LabelX1.Name = "LabelX1"
            size2 = New Size(&H12A, &H17)
            Me.LabelX1.Size = size2
            Me.LabelX1.TabIndex = 2
            Me.LabelX1.Text = "Please select the type of test station that you are working at."
            Me.AcceptButton = Me.OK_Button
            Dim ef2 As New SizeF(6!, 13!)
            Me.AutoScaleDimensions = ef2
            Me.AutoScaleMode = AutoScaleMode.Font
            Me.CancelButton = Me.Cancel_Button
            size2 = New Size(&H143, &H73)
            Me.ClientSize = size2
            Me.Controls.Add(Me.LabelX1)
            Me.Controls.Add(Me.cboTestStations)
            Me.Controls.Add(Me.TableLayoutPanel1)
            Me.DoubleBuffered = True
            Me.FormBorderStyle = FormBorderStyle.FixedDialog
            Me.MaximizeBox = False
            Me.MinimizeBox = False
            Me.Name = "dlgTestStationType"
            Me.ShowInTaskbar = False
            Me.StartPosition = FormStartPosition.CenterParent
            Me.Text = "Test Station Selection"
            Me.TableLayoutPanel1.ResumeLayout(False)
            Me.ResumeLayout(False)
        End Sub

        Private Sub OK_Button_Click(ByVal sender As Object, ByVal e As EventArgs)
            Me.DialogResult = DialogResult.OK
            Me._TestStationType = DirectCast(Me.cboTestStations.SelectedIndex, TestStationTypes)
            Me.Close
        End Sub


        ' Properties
        Friend Overridable Property TableLayoutPanel1 As TableLayoutPanel
            <DebuggerNonUserCode> _
            Get
                Return Me._TableLayoutPanel1
            End Get
            <MethodImpl(MethodImplOptions.Synchronized), DebuggerNonUserCode> _
            Set(ByVal WithEventsValue As TableLayoutPanel)
                Me._TableLayoutPanel1 = WithEventsValue
            End Set
        End Property

        Friend Overridable Property OK_Button As Button
            <DebuggerNonUserCode> _
            Get
                Return Me._OK_Button
            End Get
            <MethodImpl(MethodImplOptions.Synchronized), DebuggerNonUserCode> _
            Set(ByVal WithEventsValue As Button)
                Dim handler As EventHandler = New EventHandler(AddressOf Me.OK_Button_Click)
                If Not Object.ReferenceEquals(Me._OK_Button, Nothing) Then
                    RemoveHandler Me._OK_Button.Click, handler
                End If
                Me._OK_Button = WithEventsValue
                If Not Object.ReferenceEquals(Me._OK_Button, Nothing) Then
                    AddHandler Me._OK_Button.Click, handler
                End If
            End Set
        End Property

        Friend Overridable Property Cancel_Button As Button
            <DebuggerNonUserCode> _
            Get
                Return Me._Cancel_Button
            End Get
            <MethodImpl(MethodImplOptions.Synchronized), DebuggerNonUserCode> _
            Set(ByVal WithEventsValue As Button)
                Dim handler As EventHandler = New EventHandler(AddressOf Me.Cancel_Button_Click)
                If Not Object.ReferenceEquals(Me._Cancel_Button, Nothing) Then
                    RemoveHandler Me._Cancel_Button.Click, handler
                End If
                Me._Cancel_Button = WithEventsValue
                If Not Object.ReferenceEquals(Me._Cancel_Button, Nothing) Then
                    AddHandler Me._Cancel_Button.Click, handler
                End If
            End Set
        End Property

        Friend Overridable Property cboTestStations As ComboBoxEx
            <DebuggerNonUserCode> _
            Get
                Return Me._cboTestStations
            End Get
            <MethodImpl(MethodImplOptions.Synchronized), DebuggerNonUserCode> _
            Set(ByVal WithEventsValue As ComboBoxEx)
                Dim handler As EventHandler = New EventHandler(AddressOf Me.cboTestStations_SelectedIndexChanged)
                If Not Object.ReferenceEquals(Me._cboTestStations, Nothing) Then
                    RemoveHandler Me._cboTestStations.SelectedIndexChanged, handler
                End If
                Me._cboTestStations = WithEventsValue
                If Not Object.ReferenceEquals(Me._cboTestStations, Nothing) Then
                    AddHandler Me._cboTestStations.SelectedIndexChanged, handler
                End If
            End Set
        End Property

        Friend Overridable Property LabelX1 As LabelX
            <DebuggerNonUserCode> _
            Get
                Return Me._LabelX1
            End Get
            <MethodImpl(MethodImplOptions.Synchronized), DebuggerNonUserCode> _
            Set(ByVal WithEventsValue As LabelX)
                Me._LabelX1 = WithEventsValue
            End Set
        End Property

        Private Property _TestStationType As TestStationTypes
            <DebuggerNonUserCode> _
            Get
                Return Me.__TestStationType
            End Get
            <DebuggerNonUserCode> _
            Set(ByVal AutoPropertyValue As TestStationTypes)
                Me.__TestStationType = AutoPropertyValue
            End Set
        End Property

        Public ReadOnly Property TestStationType As TestStationTypes
            Get
                Return Me._TestStationType
            End Get
        End Property


        ' Fields
        Private Shared __ENCList As List(Of WeakReference) = New List(Of WeakReference)
        Private components As IContainer
        <AccessedThroughProperty("TableLayoutPanel1")> _
        Private _TableLayoutPanel1 As TableLayoutPanel
        <AccessedThroughProperty("OK_Button")> _
        Private _OK_Button As Button
        <AccessedThroughProperty("Cancel_Button")> _
        Private _Cancel_Button As Button
        <AccessedThroughProperty("cboTestStations")> _
        Private _cboTestStations As ComboBoxEx
        <AccessedThroughProperty("LabelX1")> _
        Private _LabelX1 As LabelX
        <DebuggerBrowsable(DebuggerBrowsableState.Never), CompilerGenerated> _
        Private __TestStationType As TestStationTypes
    End Class
End Namespace

